src/components/lesson-creation-modal.js

```diff
@@ -0,0 +1,151 @@
+  const { dateTimeFormat } = useDateFormat();
+  const { language: uiLanguage } = useLanguage();
+  const lessonApiClient = useService(LessonApiClient);
+  const { t } = useTranslation('lessonCreationModal');
+
+  const [loading, setLoading] = useState(false);
+
+  const titleValidationRules = [
+    {
+      required: true,
+      message: t('titleRequired'),
+      whitespace: true
+    }
+  ];
+
+  const slugValidationRules = [
+    {
+      validator: (rule, value) => {
+        return value && !inputValidators.isValidSlug(value)
+          ? Promise.reject(new Error(t('common:invalidSlug')))
+          : Promise.resolve();
+      }
+    }
+  ];
+
+  const defaultLesson = {
+    title: t('newLesson'),
+    slug: '',
+    language: getDefaultLanguageFromUiLanguage(uiLanguage)
+  };
+
+  useEffect(() => {
+    if (isVisible && formRef.current) {
+      formRef.current.resetFields();
+    }
+  }, [isVisible]);
+
+  const handleOk = () => {
+    if (formRef.current) {
+      formRef.current.submit();
+    }
+  };
+
+  const handleOnFinish = async ({ title, slug, language, startsOn }) => {
+    try {
+      setLoading(true);
+
+      const newLesson = await lessonApiClient.addLesson({
+        title,
+        slug: slug || '',
+        language,
+        schedule: startsOn ? { startsOn: startsOn.toDate() } : null
+      });
+
+      setLoading(false);
+      onClose();
+
+      window.location = urls.getLessonUrl(newLesson._id);
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      setLoading(false);
+    }
+  };
+
+  const handleCancel = () => onClose();
+
+  const disabledDate = current => {
+    const yesterday = new Date();
+    yesterday.setDate(yesterday.getDate() - 1);
+
+    return current && current <= yesterday;
+  };
+
+  const disabledTime = () => {
+    const disabledMorningHours = [...Array(24).keys()].splice(0, 7);
+    const disabledEveningHours = [...Array(24).keys()].splice(21);
+    const disabledMinutes = [...Array(60).keys()].filter(minute => minute % 5 !== 0);
+
+    return {
+      disabledHours: () => [...disabledMorningHours, ...disabledEveningHours],
+      disabledMinutes: () => disabledMinutes
+    };
+  };
+
+  return (
+    <Modal
+      title={t('newLesson')}
+      onOk={handleOk}
+      onCancel={handleCancel}
+      maskClosable={false}
+      visible={isVisible}
+      okButtonProps={{ loading }}
+      >
+      <Form onFinish={handleOnFinish} name="new-lesson-form" ref={formRef} layout="vertical" initialValues={defaultLesson}>
+        <FormItem label={t('common:title')} name="title" rules={titleValidationRules}>
+          <Input />
+        </FormItem>
+        <FormItem label={t('common:language')} name="language">
+          <LanguageSelect />
+        </FormItem>
+        <FormItem label={t('common:slug')} name="slug" rules={slugValidationRules}>
+          <Input />
+        </FormItem>
+        <FormItem label={t('startsOn')} name="startsOn">
+          <DatePicker
```

Comment:
The "Jetzt" button kinda defeats the purpose of these hours, you can pick for example 9:03, I'm not sure what the intention here is though. 
Maybe it could be made to pick the next closest "allowed" time?

Reply:
The "jetzt" is an out of the box default feature.
To me it makes sense as a "reset", if I select a date far in the future and then I just want to jump back to today, I click "jetzt". Not sure how the users would use it though, if at all, so for now I'd leave it until we gather feedback.